### PR TITLE
cryptsetup: 2.5.0 -> 2.6.1

### PR DESCRIPTION
**Pull request by mattmelling**
###### Description of changes

Update to 2.6.1.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by yu-re-ka — APPROVED** [review]

**Comment by NickCao** [comment]
@ofborg test systemd-initrd-luks-tpm2

File: pkgs/os-specific/linux/cryptsetup/default.nix [hunk 1]
```diff
@@ -10,14 +10,14 @@
...
-  version = "2.5.0";
+  version = "2.6.0";
 
   outputs = [ "bin" "out" "dev" "man" ];
   separateDebugInfo = true;
 
   src = fetchurl {
-    url = "mirror://kernel/linux/utils/cryptsetup/v2.5/${pname}-${version}.tar.xz";
-    sha256 = "sha256-kYSm672c5+shEVLn90GmyC8tHMDiSoTsnFKTnu4PBUI=";
+    url = "mirror://kernel/linux/utils/cryptsetup/v2.6/${pname}-${version}.tar.xz";
```
Comment (misuzu) [inline pkgs/os-specific/linux/cryptsetup/default.nix:19]:
```suggestion
    url = "mirror://kernel/linux/utils/cryptsetup/v${lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
```

File: pkgs/os-specific/linux/cryptsetup/default.nix [hunk 2]
```diff
@@ -10,14 +10,14 @@
 
 stdenv.mkDerivation rec {
   pname = "cryptsetup";
-  version = "2.5.0";
+  version = "2.6.0";
```
Comment (misuzu) [inline pkgs/os-specific/linux/cryptsetup/default.nix:13]: 2.6.1 is available https://gitlab.com/cryptsetup/cryptsetup/-/tags/v2.6.1

File: pkgs/os-specific/linux/cryptsetup/default.nix [hunk 3]
```diff
@@ -75,7 +72,7 @@ stdenv.mkDerivation rec {
     homepage = "https://gitlab.com/cryptsetup/cryptsetup/";
```
Comment (misuzu) [inline pkgs/os-specific/linux/cryptsetup/default.nix:72]:
```suggestion
    homepage = "https://gitlab.com/cryptsetup/cryptsetup/";
    changelog = "https://gitlab.com/cryptsetup/cryptsetup/-/raw/v${version}/docs/v${version}-ReleaseNotes";
```

**Review by misuzu — APPROVED** [review]
Tested after cherry-picking to b6e8ebef1966ec47edc3f1e92f3ccf6f82d0c7c4:

* builds on aarch64-linux
* fixes build on [armv7l-linux](https://hydra.armv7l.xyz/build/3882)
* systemd-initrd-luks-tpm2 test is passing on x86_64-linux

**Pull request merged**